src/common/lib/python/common_extension.c

```diff
@@ -317,6 +319,20 @@ static int PyTask_init(PyTask *self, PyObject *args, PyObject *kwds) {
     }
   }
   utarray_free(val_repr_ptrs);
+  /* Set the resource vector of the task. */
+  if (resource_vector != NULL) {
```

Comment:
I suppose we could check if requested CPU is < 1, but I know for a fact that in the Google cluster it is entirely possible to request fractional cores. The workload may be entirely memory bound, for instance, in which case it doesn't need the whole core to be dedicated to it. Fairness of these allocations would have to be a feature we add on the backend when we actually enable attribution of tasks (and , therefore, resources) to users (or drivers)